hack/generate/samples/internal/go/memcached-with-webhooks/memcached_with_webhooks.go

```diff
@@ -787,21 +829,170 @@ import (
+	Type string
+}
+
+// metricsDescription is a map of string keys (metrics) to MetricDescription values (Name, Help).
+var metricDescription = map[string]MetricDescription{
+	"MemcachedDeploymentSizeUndesiredCountTotal": {
+		Name: "memcached_deployment_size_undesired_count_total",
+		Help: "Total number of times the deployment size was not as desired.",
+		Type: "Counter",
+	},
+}
+
 var (
 	// MemcachedDeploymentSizeUndesiredCountTotal will count how many times was required
 	// to perform the operation to ensure that the number of replicas on the cluster
 	// is the same as the quantity desired and specified via the custom resource size spec.
 	MemcachedDeploymentSizeUndesiredCountTotal = prometheus.NewCounter(
 		prometheus.CounterOpts{
-			Name: "memcached_deployment_size_undesired_count_total",
-			Help: "Total number of times the deployment size was not as desired.",
+			Name: metricDescription["MemcachedDeploymentSizeUndesiredCountTotal"].Name,
+			Help: metricDescription["MemcachedDeploymentSizeUndesiredCountTotal"].Help,
 		},
 	)
 )
-// Register metrics with the global prometheus registry
+
+// RegisterMetrics will register metrics with the global prometheus registry
 func RegisterMetrics() {
 	metrics.Registry.MustRegister(MemcachedDeploymentSizeUndesiredCountTotal)
 }
+
+// ListMetrics will create a slice with the metrics available in metricDescription
+func ListMetrics() []MetricDescription {
+	v := make([]MetricDescription, 0, len(metricDescription))
+	// Insert value (Name, Help) for each metric
+	for _, value := range metricDescription {
+		v = append(v, value)
+	}
+
+	return v
+}
+`
+
+const metricsdocsFragment = `
+
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/example/memcached-operator/monitoring"
+)
+
+// please run \"make generate-metricsdocs\" to generate this document.
+// constant parts of the file
+const (
+	title      = "# Memcached Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the memcached-operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool \"monitoring/metricsdocs\" and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## Memcached Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run \"make generate-metricsdocs\" to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change \"monitoring/metricsdocs\" according to your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// TODO: scaffolding these helpers with operator-lib: https://github.com/operator-framework/operator-lib.
+
+// MemcachedMetricList contains the name, description, and type for each metric.
+func main() {
+	MemcachedMetricList := metricDescriptionListToMetricList(monitoring.ListMetrics())
```

Comment:
@avlitman I think we should replace the "Memcached" with "Operator" in all places, so that developers can copy and use this example.

Reply:
Except for the metric name off course

Reply:
Agree

Reply:
updated